tensorflow/python/eager/backprop.py

```diff
@@ -811,6 +827,9 @@ def watch(self, tensor):
     Args:
       tensor: a Tensor or list of Tensors.
     """
+    if check_float_dtype(tensor) == False:
```

Comment:
You can fuse `check_float_dtype` into the for loop. This will allow to issue a more specific warning (see comment below):

```python
for t in tf.nest.flatten(tensor):
  if not t.dtype.is_floating:
    ...
```

Reply:
Thanks for your review. Done now.